sdk/cognitiveservices/textanalytics/Azure.AI.TextAnalytics/src/TextAnalyticsClient.cs

```diff
@@ -0,0 +1,1422 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using Azure.Core;
+using Azure.Core.Pipeline;
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Azure.AI.TextAnalytics
+{
+    /// <summary>
+    /// The client to use for interacting with the Azure Configuration Store.
+    /// </summary>
+    public partial class TextAnalyticsClient
+    {
+        private readonly Uri _baseUri;
+        private readonly HttpPipeline _pipeline;
+        private readonly ClientDiagnostics _clientDiagnostics;
+        private readonly string _subscriptionKey;
+        private readonly string _apiVersion;
+
+        /// <summary>
+        /// Protected constructor to allow mocking.
+        /// </summary>
+        protected TextAnalyticsClient()
+        {
+        }
+
+        /// <summary>
+        /// </summary>
+        /// <param name="endpoint"></param>
+        /// <param name="credential"></param>
+        public TextAnalyticsClient(Uri endpoint, TokenCredential credential)
+            : this(endpoint, credential, new TextAnalyticsClientOptions())
+        {
+        }
+
+        /// <summary>
+        /// </summary>
+        /// <param name="endpoint"></param>
+        /// <param name="credential"></param>
+        /// <param name="options"></param>
+        public TextAnalyticsClient(Uri endpoint, TokenCredential credential, TextAnalyticsClientOptions options)
```

Comment:
In KV, we actually allow `options` to be null but create it if necessary, thus not throwing an exception:

```csharp
options ??= new TextAnalyticsOptions();
```